ServiceBus/ServiceBus/ProductService.svc.cs

```diff
@@ -136,63 +141,74 @@ public Result AddProduct(string name, double price, Guid guid)
         /// <param name="guid">ID of a product</param>
         /// <param name="name">New name for a product</param>
         /// <param name="price">New price for a product</param>
+        /// <param name="pType">Type of a product</param>
         /// <returns>Modified product</returns>
-        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price)
+        public SharedLibs.DataContracts.Product EditProduct(Guid guid, string name, double price, ProductType pType = null, bool enabled = true, bool headliner = false)
         {
             try
             {
-                var originalProduct = GetProduct(guid);
-
-                if (originalProduct.Result.ResultType == ResultType.Success)

```

Comment:
I have changed this part because method get guid then it search for DTO product then new object (context) for product is created then it is populated with data from DTO product object and then some modification could begin.

Now it should be get guid, find guid in db and if found create object filled with database data.

Goal was to simplify the method.